src/controllers/controllerhidreporttabsmanager.cpp

```diff
@@ -0,0 +1,488 @@
+    }
+
+    updateTableWithReportData(table, reportData);
+}
+
+void ControllerHidReportTabsManager::slotSendReport(QTableWidget* table,
+        quint8 reportId,
+        hid::reportDescriptor::HidReportType reportType) {
+    if (!m_pHidController->isOpen()) {
+        qWarning() << "HID controller is not open.";
+        return;
+    }
+
+    HidControllerJSProxy* jsProxy = static_cast<HidControllerJSProxy*>(m_pHidController->jsProxy());
+    VERIFY_OR_DEBUG_ASSERT(jsProxy) {
+        return;
+    }
+
+    const auto& reportDescriptor = *m_pHidController->getReportDescriptor();
+
+    auto report = reportDescriptor.getReport(reportType, reportId);
+    VERIFY_OR_DEBUG_ASSERT(report) {
+        return;
+    }
+
+    // Create a QByteArray of the size of the report
+    QByteArray reportData(report->getReportSize(), 0);
+
+    // Iterate through each row in the table
+    for (int row = 0; row < table->rowCount(); ++row) {
+        auto* item = table->item(row, 5); // Value column is at index 5
+        if (item) {
+            // Retrieve custom data from the first cell
+            QVariant customData = table->item(row, 0)->data(Qt::UserRole + 1);
+            if (customData.isValid()) {
+                auto control =
+                        reinterpret_cast<hid::reportDescriptor::Control*>(
+                                customData.value<void*>());
+                // Set the control value in the reportData
+                bool success = hid::reportDescriptor::applyLogicalValue(
+                        reportData, *control, item->text().toLongLong());
+                if (!success) {
+                    qWarning() << "Failed to set control value for row" << row;
+                    continue;
+                }
+            }
+        }
+    }
+
+    // Send the reportData
+    if (reportType == hid::reportDescriptor::HidReportType::Feature) {
+        jsProxy->sendFeatureReport(reportId, reportData);
+    } else if (reportType == hid::reportDescriptor::HidReportType::Output) {
+        jsProxy->sendOutputReport(reportId, reportData);
+    }
+}
+
+void ControllerHidReportTabsManager::populateHidReportTable(
+        QTableWidget* table,
+        const hid::reportDescriptor::Report& report,
+        hid::reportDescriptor::HidReportType reportType) {
+    // Temporarily disable updates to speed up populating
+    table->setUpdatesEnabled(false);
+
+    // Reserve rows up-front
+    const auto& controls = report.getControls();
+    table->setRowCount(static_cast<int>(controls.size()));
+
+    // Set the delegate once if needed
+    if (reportType != hid::reportDescriptor::HidReportType::Input) {
+        table->setItemDelegateForColumn(5, new ValueItemDelegate(table));
+    }
+
+    bool showVolatileColumn = (reportType == hid::reportDescriptor::HidReportType::Feature ||
+            reportType == hid::reportDescriptor::HidReportType::Output);
+
+    // Set headers
+    QStringList headers = {QStringLiteral("Byte Position"),
```

Comment:
These are user facing strings, right? Than please use tr() to make them translatable.